### PR TITLE
[OSJC-140] Replace use of enums where it makes sense

### PR DESCRIPTION
**Pull request by jcantrill**
@adietish Please review and merge.

### CODE REVIEW

**Comment by adietish** [comment]
+1 looks good

File: src/main/java/com/openshift/restclient/model/build/BuildStrategyType.java [hunk 1]
```diff
@@ -13,11 +13,14 @@
  * by OpenShift
  */
 
-// TODO: evalute switching to a class/constants since enums are not extendable
-public enum BuildStrategyType {
-	Docker,
+public interface BuildStrategyType {
+	
+	static final String Docker = "Docker";
```
Comment (adietish) [inline src/main/java/com/openshift/restclient/model/build/BuildStrategyType.java]: Something I didnt catch so far: 
Shouldnt name of public (final) constants be capitalized? http://stackoverflow.com/a/7259738/231357

**Comment by jcantrill** [comment]
@adietish Updated to follow the typical java convention.

**Comment by adietish** [comment]
ok, looks good to me. +1

File: src/main/java/com/openshift/internal/restclient/model/BuildConfig.java [hunk 2]
```diff
@@ -144,7 +146,7 @@ public String getSourceURI() {
 	@Override
 	public void setBuildSource(IBuildSource source){
 		switch(source.getType()) {
-		case Git:
+		case BuildSourceType.Git:
```
Comment (adietish) [inline src/main/java/com/openshift/internal/restclient/model/BuildConfig.java]: Looks like you missed these constants?
Are there others?

File: src/main/java/com/openshift/internal/restclient/DefaultClient.java [hunk 3]
```diff
@@ -66,7 +66,7 @@
 	private static final String apiEndpoint = "api";
 	private static final String osApiEndpoint = "osapi";
```
Comment (adietish) [inline src/main/java/com/openshift/internal/restclient/DefaultClient.java]: arent these constants, too that we should capitalize?

File: src/main/java/com/openshift/restclient/model/build/BuildSourceType.java [hunk 4]
```diff
@@ -11,7 +11,8 @@
 /**
  * @author Jeff Cantrill
  */
-// TODO: evaluate switching this enum to a class since enums are not extendable
-public enum BuildSourceType {
-	Git
+public interface BuildSourceType {
+	
+	static final String Git = "Git";
```
Comment (adietish) [inline src/main/java/com/openshift/restclient/model/build/BuildSourceType.java]: constant?

File: src/main/java/com/openshift/restclient/model/build/BuildStrategyType.java (hunk 1 above)
Comment (adietish) [inline src/main/java/com/openshift/restclient/model/build/BuildStrategyType.java]: constant?

File: src/main/java/com/openshift/restclient/model/build/BuildStrategyType.java [hunk 5]
```diff
@@ -13,11 +13,14 @@
...
+	
 	@Deprecated
-	STI,
-	Source,
-	Custom
+	static final String STI = "STI";
+	
+	static final String Source = "Source";
+	static final String Custom = "Custom";
```
Comment (adietish) [inline src/main/java/com/openshift/restclient/model/build/BuildStrategyType.java]: these 2 constants, too, no?

File: src/main/java/com/openshift/restclient/model/build/BuildTriggerType.java [hunk 6]
```diff
@@ -11,17 +11,18 @@
...
 	@Deprecated
-	imageChange,
+	static final String imageChange = "imageChange";
 
-	Generic,
-	GitHub,
-	ImageChange;
+	static final String Generic = "Generic";
+	static final String GitHub = "GitHub";
+	static final String ImageChange = "ImageChange";
```
Comment (adietish) [inline src/main/java/com/openshift/restclient/model/build/BuildTriggerType.java]: dito?

File: src/main/java/com/openshift/restclient/model/route/TLSTerminationType.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * 
+ * @author jeff.cantrill
+ *
+ */
+public interface TLSTerminationType {
+	static final String edge = "edge";
+	static final String passthrough = "passthrough";
+	static final String reencrypt = "reencrypt";
```
Comment (adietish) [inline src/main/java/com/openshift/restclient/model/route/TLSTerminationType.java]: these?

**Comment by adietish** [comment]
I did a search for "static final" and found a few more that we didnt correct. I guess that I merge this PR and file a jira to correct the other ones, agree?

**Comment by adietish** [comment]
I created https://issues.jboss.org/browse/OSJC-181 and https://issues.jboss.org/browse/JBIDE-20122

**Comment by adietish** [comment]
merged and pushed to upstream/master.
and published to maven repository.

**Pull request merged**